modules/tinymce/src/plugins/link/test/ts/browser/SelectedImageLinkTest.ts

```diff
@@ -1,76 +1,72 @@
+import { SugarBody, SugarDocument } from '@ephox/sugar';
+import Editor from 'tinymce/core/api/Editor';
+import Plugin from 'tinymce/plugins/link/Plugin';
+import Theme from 'tinymce/themes/silver/Theme';
 
-  TinyLoader.setupLight((editor, onSuccess, onFailure) => {
-    const tinyApis = TinyApis(editor);
-    const tinyUi = TinyUi(editor);
-    const doc = TinyDom.fromDom(document);
+import { TestLinkUi } from '../module/TestLinkUi';
 
-    Pipeline.async({}, [
-      TestLinkUi.sClearHistory,
-      Log.stepsAsStep('TBA', 'Link: images should be preserved when adding a link', [
-        tinyApis.sSetContent('<p><img src="image.png"></p>'),
-        tinyApis.sSelect('img', []),
-        TestLinkUi.sOpenLinkDialog(tinyUi),
-        FocusTools.sSetActiveValue(doc, 'http://something'),
-        UiFinder.sNotExists(SugarBody.body(), '.tox-label:contains("Text to display")'),
-        TestLinkUi.sClickSave,
-        Waiter.sTryUntil(
-          'Wait until link is inserted',
-          tinyApis.sAssertContentPresence({
-            'a[href="http://something"]': 1,
-            'img[src="image.png"]': 1,
-            'p': 1
-          })
-        )
-      ]),
-      Log.stepsAsStep('TBA', 'Link: images should be preserved when editing a link', [
-        tinyApis.sSetContent('<p><a href="http://www.google.com/"><img src="image.png"></a></p>'),
-        tinyApis.sSelect('a', []),
-        TestLinkUi.sOpenLinkDialog(tinyUi),
-        FocusTools.sSetActiveValue(doc, 'http://something'),
-        UiFinder.sNotExists(SugarBody.body(), '.tox-label:contains("Text to display")'),
-        TestLinkUi.sClickSave,
-        Waiter.sTryUntil(
-          'Wait until link is updated',
-          tinyApis.sAssertContentPresence({
-            'a[href="http://something"]': 1,
-            'img[src="image.png"]': 1,
-            'p': 1
-          })
-        )
-      ]),
-      Log.stepsAsStep('TINY-4706', 'Link: images link urls should be able to be removed', [
-        tinyApis.sSetContent('<p><a href="http://www.google.com/" title="test"><img src="image.png"></a></p>'),
-        tinyApis.sSelect('a', []),
-        TestLinkUi.sOpenLinkDialog(tinyUi),
-        Chain.asStep(SugarBody.body(), [
-          FocusTools.cSetActiveValue(''),
-          TestLinkUi.cFireEvent('input')
-        ]),
-        UiFinder.sNotExists(SugarBody.body(), '.tox-label:contains("Text to display")'),
-        TestLinkUi.sAssertDialogContents({
-          url: '',
-          title: 'test'
-        }),
-        TestLinkUi.sClickCancel
-      ]),
-      TestLinkUi.sClearHistory
-    ], onSuccess, onFailure);
-  }, {
+describe('browser.tinymce.plugins.link.SelectedImageTest', () => {
+  const hook = TinyHooks.bddSetupLight<Editor>({
     plugins: 'link',
     toolbar: 'link',
-    theme: 'silver',
     base_url: '/project/tinymce/js/tinymce'
-  }, success, failure);
+  }, [ Plugin, Theme ]);
+
+  const doc = SugarDocument.getDocument();
+
+  before(() => {
```

Comment:
eta reduce please

Reply:
there are a few other instances of this later in the PR, I'm going to stop commenting on them now

Reply:
As mentioned above, we shouldn't be ETA reducing hooks.